flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/StreamTableSourceTable.scala

```diff
@@ -39,6 +39,9 @@ class StreamTableSourceTable[T](
     val fieldCnt = fieldNames.length
 
     val rowtime = tableSource match {
+      case nullTimeSource : DefinedRowtimeAttribute if nullTimeSource.getRowtimeAttribute == null => None
+      case emptyStringTimeSource : DefinedRowtimeAttribute if emptyStringTimeSource.getRowtimeAttribute.trim.equals("")  =>
+        throw TableException("The name of the rowtime attribute must not be empty.")
       case timeSource: DefinedRowtimeAttribute if timeSource.getRowtimeAttribute != null =>
```

Comment:
You mean there is no need to add the last if condition (should change "case timeSource: DefinedRowtimeAttribute if timeSource.getRowtimeAttribute != null =>" to "case timeSource: DefinedRowtimeAttribute =>")? Is it right?

Thanks for your patient guidance.

I commited again.